docs/cheatsheet.rst

```diff
@@ -90,7 +90,7 @@ Global Variables
 - ``block.difficulty`` (``uint``): current block difficulty
 - ``block.gaslimit`` (``uint``): current block gaslimit
 - ``block.number`` (``uint``): current block number
-- ``block.timestamp`` (``uint``): current block timestamp
+- ``block.timestamp`` (``uint``): current block timestamp of the block where the transaction was mined; for no-transaction calls this value can be anything from zero to a random number depending on the EVM implementation
```

Comment:
If we do that much in detail here (which I think we shouldn't) then we should also do the same change to all the other `block` members.

Reply:
Wouldn't a "warning"/"notice" block be enough for explaining that these parameters can have arbitrary value in case of off-chain evaluation?

Reply:
Happy to amend the pull request and make it a generic warning for all `block` members.

Reply:
@miohtama You could add it as a new note before the note "Do not rely on ``block.timestamp`` or ``blockhash`` as a source of randomness, unless you know what you are doing."